### ISSUE TITLE
Unlatch - Failed to perform the action button/press. 'NukiButton' object has no attribute 'entity_description'

### ISSUE THREAD

**Issue by RambaaZambaa** [ronengr/hass_nuki_bt#84]
## System Information

version | core-2024.11.1
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.31-haos-raspi
arch | aarch64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 5000
Installed Version | 2.0.1
Stage | running
Available Repositories | 1461
Downloaded Repositories | 10

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | January 1, 2018 at 01:00
relayer_connected | false
relayer_region | null
remote_enabled | false
remote_connected | false
alexa_enabled | true
google_enabled | true
cloud_ice_servers_enabled | true
remote_server | null
certificate_status | null
instance_id | 96b5967bfaf14b7b90b31171f1a33525
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 13.2
-- | --
update_channel | stable
supervisor_version | supervisor-2024.11.2
agent_version | 1.6.0
docker_version | 27.2.0
disk_total | 57.8 GB
disk_used | 25.1 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | 
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.8.0), Mosquitto broker (6.4.1), TasmoAdmin (0.30.6), Terminal & SSH (9.15.0), ESPHome (2024.10.3), Cloudflared (5.2.1), Home Assistant Google Drive Backup (0.112.1), Music Assistant Server (2.3.2)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 3
views | 4
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | November 6, 2024 at 15:36
-- | --
current_recorder_run | November 12, 2024 at 16:31
estimated_db_size | 581.78 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/ronengr/hass_nuki_bt/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Nuki Device type

Nuki Smart Lock 4. Generation (non Pro) - Firmware: 4.4.8

### hass_nuki_bt version

0.0.15.1

### Describe the issue

The **unlatch** control ist not working => Failed to perform the action button/press. 'NukiButton' object has no attribute 'entity_description'
Other controls like **lock** are working.
Control via Nuki-App is also working.

### Reproduction steps

1. Press Unlatch
2. Error "Failed to perform the action button/press. 'NukiButton' object has no attribute 'entity_description'"

### Debug logs

```text
2024-11-13 18:35:07.068 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration ble_monitor which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:35:07.073 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:35:07.076 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:35:07.080 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration thermal_comfort which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:35:07.092 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hifiberry which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:35:07.095 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hass_nuki_bt which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:35:07.097 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration spotcast which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:35:25.954 ERROR (ImportExecutor_0) [homeassistant] Error doing job: Unclosed client session (None)
2024-11-13 18:35:33.132 WARNING (zeroconf-ServiceBrowser-_googlecast._tcp-120) [pychromecast.dial] Failed to determine cast type for host <unknown> (<urlopen error [Errno 111] Connection refused>) (services:{MDNSServiceInfo(name='Baggen-XL-f62ae76036a03d499a1f19f96e8b3c07._googlecast._tcp.local.')})
2024-11-13 18:35:33.207 WARNING (zeroconf-ServiceBrowser-_googlecast._tcp-120) [pychromecast.dial] Failed to determine cast type for host <unknown> (<urlopen error [Errno 111] Connection refused>) (services:{MDNSServiceInfo(name='Stammen-L-ea69781405647338db1b0f9e7482f7d7._googlecast._tcp.local.')})
2024-11-13 18:35:33.801 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command KEYTURNER_STATES. Got 1 unknown bytes with value: b'\x00'
2024-11-13 18:35:33.968 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command CONFIG. Got 1 unknown bytes with value: b'\x01'
2024-11-13 18:35:34.046 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command LOG_ENTRY. Got 3 unknown bytes with value: b'\x04\x04\x08'
2024-11-13 18:35:34.185 WARNING (MainThread) [aiowithings] 94 is an unsupported value for <enum 'DeviceModel'>, please report this at https://github.com/joostlek/python-withings/issues
2024-11-13 18:35:34.904 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command KEYTURNER_STATES. Got 1 unknown bytes with value: b'\x00'
2024-11-13 18:35:35.027 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command LOG_ENTRY. Got 3 unknown bytes with value: b'\x04\x04\x08'
2024-11-13 18:35:56.665 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547829728592] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2802, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2845, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
    await self.async_press()
  File "/config/custom_components/hass_nuki_bt/button.py", line 82, in async_press
    await self.entity_description._action_function(self)
          ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NukiButton' object has no attribute 'entity_description'
2024-11-13 18:36:26.054 ERROR (MainThread) [homeassistant] Error doing job: Unclosed client session (None)
2024-11-13 18:37:27.565 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547829728592] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2802, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2845, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
    await self.async_press()
  File "/config/custom_components/hass_nuki_bt/button.py", line 82, in async_press
    await self.entity_description._action_function(self)
          ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NukiButton' object has no attribute 'entity_description'
```

**Comment by ronengr** [comment]
Thanks for the feedback.
I will look into this.

**Comment by RambaaZambaa** [comment]
Updated to 0.0.16.1 (was not showing, because HACS was not loaded because i forgot to clear the browsercache after HA-Update).
Same error.

New Debug-Log:
```
2024-11-13 18:48:17.704 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration ble_monitor which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:48:17.707 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:48:17.709 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:48:17.711 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration thermal_comfort which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:48:17.712 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hifiberry which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:48:17.714 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hass_nuki_bt which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:48:17.716 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration spotcast which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-13 18:48:34.370 ERROR (ImportExecutor_0) [homeassistant] Error doing job: Unclosed client session (None)
2024-11-13 18:48:37.591 WARNING (zeroconf-ServiceBrowser-_googlecast._tcp-146) [pychromecast.dial] Failed to determine cast type for host <unknown> (<urlopen error [Errno 111] Connection refused>) (services:{MDNSServiceInfo(name='Baggen-XL-f62ae76036a03d499a1f19f96e8b3c07._googlecast._tcp.local.')})
2024-11-13 18:48:37.650 WARNING (zeroconf-ServiceBrowser-_googlecast._tcp-146) [pychromecast.dial] Failed to determine cast type for host <unknown> (<urlopen error [Errno 111] Connection refused>) (services:{MDNSServiceInfo(name='Stammen-L-ea69781405647338db1b0f9e7482f7d7._googlecast._tcp.local.')})
2024-11-13 18:48:37.912 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command KEYTURNER_STATES. Got 1 unknown bytes with value: b'\x00'
2024-11-13 18:48:38.027 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command CONFIG. Got 1 unknown bytes with value: b'\x01'
2024-11-13 18:48:38.102 WARNING (MainThread) [pyNukiBT.nuki] Got unexpected message length for command LOG_ENTRY. Got 3 unknown bytes with value: b'\x04\x04\x08'
2024-11-13 18:48:38.392 WARNING (MainThread) [aiowithings] 94 is an unsupported value for <enum 'DeviceModel'>, please report this at https://github.com/joostlek/python-withings/issues
2024-11-13 18:48:51.387 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547177626880] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2802, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2845, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
    await self.async_press()
  File "/config/custom_components/hass_nuki_bt/button.py", line 82, in async_press
    await self.entity_description._action_function(self)
          ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NukiButton' object has no attribute 'entity_description'
2024-11-13 18:49:36.705 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547177626880] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2802, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2845, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
    await self.async_press()
  File "/config/custom_components/hass_nuki_bt/button.py", line 82, in async_press
    await self.entity_description._action_function(self)
          ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NukiButton' object has no attribute 'entity_description'
2024-11-13 18:49:43.051 ERROR (SyncWorker_33) [homeassistant] Error doing job: Unclosed client session (None)
2024-11-13 18:49:59.144 ERROR (SyncWorker_26) [homeassistant] Error doing job: Unclosed client session (None)
2024-11-13 18:51:39.817 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547177626880] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2802, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2845, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
    await self.async_press()
  File "/config/custom_components/hass_nuki_bt/button.py", line 82, in async_press
    await self.entity_description._action_function(self)
          ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NukiButton' object has no attribute 'entity_description'
```

**Comment by ronengr** [comment]
@RambaaZambaa 
Thanks for the feedback.

Fixed in version 0.0.16.2

**Comment by RambaaZambaa** [comment]
Works. Thank you very much! That was lightning fast :)
